src/main/java/io/neonbee/entity/EntityVerticle.java

```diff
@@ -250,34 +258,65 @@ private Future<Void> announceEntityVerticle(Vertx vertx) {
-                                                        "Announce entity {} is served by entity verticle with qualified name {}",
-                                                        name, qualifiedName);
-                                                return sharedMap.put(name, qualifiedNames);
-                                            }).onComplete(anyResult -> {
-                                                LOGGER.debug("Releasing lock {}", name);
-                                                lock.release();
-                                            });
-                                        });
+                                String qualifiedName = getQualifiedName();
+                                return registry.register(name, qualifiedName, promise -> {
+                                    if (vertx.isClustered()) {
+                                        addToRegistrationList(vertx, name, qualifiedName).onComplete(promise);
+                                    } else {
+                                        promise.complete();
+                                    }
+                                });
                             }).collect(Collectors.toList());
                     return allComposite(announceFutures).mapEmpty();
                 });
     }
+
+    private static Future<Void> addToRegistrationList(Vertx vertx, String name, String qualifiedName) {
+        LOGGER.debug("Add name: {} and qualified name: {} ", name, qualifiedName);
+
+        String clusterNodeId = ClusterHelper.getNodeId(vertx);
+        LOGGER.debug("Acquire lock for cluster node ID: {}", clusterNodeId);
+        return vertx.sharedData().getLock(clusterNodeId)
+                .onFailure(cause -> LOGGER.error("Failed to acquire lock for cluster node ID: {}", clusterNodeId))
+                .compose(lock -> getNodeEntitiesMap(vertx).compose(map -> map.get(clusterNodeId)
+                        .map(jsonArray -> jsonArray == null ? new JsonArray() : (JsonArray) jsonArray)
+                        .compose(jsonArray -> {
+                            JsonObject jo = new JsonObject().put(QUALIFIED_NAME, qualifiedName).put(ENTITY_NAME, name);
+
+                            if (!jsonArray.contains(jo)) {
+                                jsonArray.add(jo);
+                                return map.put(clusterNodeId, jsonArray)
+                                        .onSuccess(event -> LOGGER.info(
+                                                "registered, cluster node ID: {}, Entity name: {}, qualified name: {} ",
+                                                clusterNodeId, name, qualifiedName));
+                            }
+                            return succeededFuture();
+                        })).onComplete(unused -> {
+                            LOGGER.debug("Release lock for cluster node ID: {}", clusterNodeId);
+                            lock.release();
+                        }));
+    }
```

Comment:
If you make *AsyncSharedMapRegistry* make also working for List<JsonObject> and not only for *String* you could reuse it. Maybe it is a little bit tricky, because in this case you directly add a whole list and not only one element into a list, but we can figure this out. I guess there is a pretty way to do this.